### ISSUE TITLE
Teleport kube access doesn't poperly pass watch updates to lens

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#30464]
### Expected behavior:

When using Lens or OpenLens, the typical behavior is that the list of resources will auto-update as updates happen.

While some types of resources, like events, do seem to update properly in lens, other resources do not.

### Current behavior:

I recreated this issue by using OpenLens: 6.5.2-366 on my intel mac.

I tried showing the list of pods in the "default" namespace when lens was connected to a k3s v1.27.4+k3s1 kube cluster -- both directly, and via kube access.

When connected via kube access, the updates would never come and lens would not update, while lens would properly update when connected directly.

I am able to run `kubectl get pod -w` and see updates happen, so I believe there is a bug that may be conditionally triggered-- maybe a buffering behavior, or an early connection close that only happens when a slightly different request is made.

I enabled the kubernetes audit feature and confirmed that the only differences between the pod watch operations as seen from the k3s cluster were the authentication method used, which is expected. The rest of the requests appear to be identical.

### Bug details:
- Teleport version - Teleport v13.3.2 git:v13.3.2-0-gfc60440 go1.20.7 (tsh, proxy, auth, and agent all on same version)
- Recreation steps
  - configure openlens or lens to use a kube context via teleport kube access.
  - open a list of pods
  - create or delete pods using a separate terminal and note that the changes don't

**Comment by tigrato** [comment]
This problem is caused by the [`oxy`](https://github.com/gravitational/oxy) dependency we used in Teleport 13, 12, and 11 that handles the forwarding of requests from the Teleport `kubernetes_service` to the Kubernetes API.

When the Kubernetes API sends the event, `oxy.Forwarder` fails to write the full payload in one operation. Instead, it writes the payload in two or more operations. 
`oxy` uses an in-memory buffer to reduce the number of write operations but this approach has a serious limitation when the payload size is bigger than the buffer. When it happens, `oxy` writes the payload split into multiple writes: 
- write 1: [0, memBufSize-1]
- write 2: [memBufSize, end]

The last write operation only happens 100ms after the previous write which creates a problem on the client side because it receives the event payload with an incomplete JSON document and terminates the connection.

For Teleport 14, we completely replaced `oxy` usage for forwarding requests and this problem is no longer happening - I tested it  and everything works correctly. 

Since replacing `oxy` was a major change on the Teleport side with huge impacts on App Access, Kubernetes Access, and Okta, it wasn't backported to the other release branches. We will re-evaluate this decision after we finish the Teleport 14 Test Plan.



https://github.com/gravitational/teleport/assets/3629062/0914c2ce-b751-4904-8c50-a433cb80631b
